CODIGO/ao20audio.bas

```diff
@@ -0,0 +1,198 @@
+'    it under the terms of the GNU Affero General Public License as published by
+'    the Free Software Foundation, either version 3 of the License, or
+'    (at your option) any later version.
+'
+'    This program is distributed in the hope that it will be useful,
+'    but WITHOUT ANY WARRANTY; without even the implied warranty of
+'    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+'    GNU Affero General Public License for more details.
+'    You should have received a copy of the GNU Affero General Public License
+'    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+'
+'
+Option Explicit
+
+Public AudioEngine As clsAudioEngine
+Public MusicEnabled As Boolean
+Public FxEnabled As Boolean
+Public AudioEnabled As Boolean
+Public AmbientEnabled As Boolean
+Private CurMusicVolume As Long
+Private CurAmbientVolume As Long
+Private CurFxVolume As Long
+
+Public Sub CreateAudioEngine(ByVal hwnd As Long, ByRef dx8 As DirectX8, ByRef renderer As clsAudioEngine)
+On Error GoTo AudioEngineInitErr:
+    If AudioEnabled Then
+        CurMusicVolume = 0
+        Set AudioEngine = New clsAudioEngine
+        Call AudioEngine.init(dx8, hwnd)
+        Debug.Print "Audio Engine OK"
+        Exit Sub
+    Else
+        Debug.Print "Warning Audio Disabled"
+    End If
+    
+    Exit Sub
+AudioEngineInitErr:
+    Call MsgBox("Error creating audio engine", vbCritical, "Argentum20")
+    Debug.Print "Error Number Returned: " & Err.Number
+    End
+End Sub
+Public Function StopAmbientAudio() As Long
+    StopAmbientAudio = -1
+    If AudioEnabled And AmbientEnabled Then
+        StopAmbientAudio = ao20audio.AudioEngine.stop_ambient
+    End If
+End Function
+Public Sub PlayAmbientAudio(ByVal UserMap As Long)
+    If AudioEnabled And AmbientEnabled Then
+        Dim wav As Integer
+        If EsNoche Then
+            wav = ReadField(1, Val(MapDat.ambient), Asc("-"))
+        Else
+            wav = ReadField(2, Val(MapDat.ambient), Asc("-"))
+            If wav = 0 Then Exit Sub
+        End If
+        Call ao20audio.AudioEngine.play_ambient(wav, True, CurAmbientVolume)
+    End If
+End Sub
+Public Function playwav(ByVal id As Integer, Optional ByVal looping As Boolean = False, Optional ByVal volume As Long = 0, Optional ByVal pan As Long = 0) As Long
+    playwav = -1
+    If AudioEnabled And FxEnabled Then
+        PlayWav = ao20audio.AudioEngine.play_wav(id, looping, volume, pan)
+    End If
+End Function
+Public Function stopwav(ByVal id As Integer) As Long
+   stopwav = -1
+    If AudioEnabled And FxEnabled Then
+        StopWav = ao20audio.AudioEngine.stop_wav(id)
+    End If
+End Function
+Public Function playmidi(ByVal id As Integer, Optional ByVal looping As Boolean = False, Optional ByVal volume As Long = 0) As Long
+    playmidi = -1
+    If AudioEnabled And MusicEnabled Then
+        PlayMidi = ao20audio.AudioEngine.play_midi(id, looping, volume)
+    End If
+End Function
+Public Sub stopallplayback()
```

Comment:
PascalCase?

it will be nice to have a space between functions

Reply:
Done in next patch